src/Model/Relations/AbstractRelation.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace Mongolid\Model\Relations;
+
+use Mongolid\Container\Ioc;
+use Mongolid\Model\DocumentEmbedder;
+use Mongolid\Model\HasAttributesInterface;
+
+abstract class AbstractRelation implements RelationInterface
+{
+    /**
+     * @var HasAttributesInterface
+     */
+    protected $parent;
+
+    /**
+     * @var string
+     */
+    protected $entity;
+
+    /**
+     * @var string
+     */
+    protected $field;
+
+    /**
+     * @var DocumentEmbedder
+     */
+    protected $documentEmbedder;
+
+    /**
+     * @var string
+     */
+    protected $relationName;
+
+    public function __construct(HasAttributesInterface $parent, string $entity, string $field, string $relationName)
+    {
+        if ($relationName === $field) {
```

Comment:
Do we need to check for some reserved words, for example, `attributes` or something like that?

Reply:
no, since the `attributes` attribute is private. so there will be no collision if we use it on the model.

Reply:
I will write a Integration test for it :wink: